src/Evolution/Systems/GeneralizedHarmonic/Equations.cpp

```diff
@@ -8,9 +8,37 @@
 #include "DataStructures/DataVector.hpp"
 #include "DataStructures/Tensor/EagerMath/DotProduct.hpp"
 #include "DataStructures/Tensor/Tensor.hpp"  // IWYU pragma: keep
+#include "Evolution/Systems/GeneralizedHarmonic/Characteristics.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
 #include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+template <typename TagsList>
+class Variables;
 
 // IWYU pragma: no_forward_declare Tensor
+namespace {
+template <typename FieldTag>
+db::item_type<FieldTag> weight_char_field(
+    const db::item_type<FieldTag>& char_field_int,
+    const DataVector& char_speed_int,
+    const db::item_type<FieldTag>& char_field_ext,
+    const DataVector& char_speed_ext) noexcept {
+  const DataVector& char_speed_avg{0.5 * (char_speed_int + char_speed_ext)};
+  db::item_type<FieldTag> weighted_char_field = char_field_int;
+  auto weighted_char_field_it = weighted_char_field.begin();
+  for (auto int_it = char_field_int.begin(), ext_it = char_field_ext.begin();
+       int_it != char_field_int.end();
+       ++int_it, ++ext_it, ++weighted_char_field_it) {
+    *weighted_char_field_it *= char_speed_avg * step_function(char_speed_avg);
+    *weighted_char_field_it +=
+        char_speed_avg * step_function(-char_speed_avg) * *ext_it;
+  }
+
+  return weighted_char_field;
+}
+}  // namespace
```

Comment:
`}  // namespace GeneralizedHarmonic_detail`

In your test you can then forward declare:
```cpp
namespace GeneralizedHarmonic_detail {
... weight_char_fields(...);
}
```